### PR TITLE
AISDK-51: Add streaming responses to SDK

### PR DESCRIPTION
**Pull request by kbridbur**
Todo:
rebase

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -112,3 +112,23 @@ Both the json and object forms contain all the formation outlined in the respons
...
+
+You can also get captions output from the SDK:
+
+```python
+captions = client.get_captions(job.id)
+```
+
+## Streamed outputs
+
+Any output format can be retrieved as a stream for convenience. In these cases we return the raw http response to you. The output can be retrieved via `response.content`, `response.iter_lines()` or `response.iter_content()`.
```
Comment (amikofalvy) [inline README.md]:
```suggestion
Any output format can be retrieved as a stream. In these cases we return the raw http response to you. The output can be retrieved via `response.content`, `response.iter_lines()` or `response.iter_content()`.
```

File: src/rev_ai/apiclient.py [hunk 2]
```diff
@@ -195,19 +193,30 @@ def get_transcript_text(self, id_, filename=None, filepath=None):
...
+    def get_transcript_text_as_stream(self, id_):
+        """Get the transcript of a specific job as a plain text stream.
+
+        :param id_: id of job to be requested
+        :returns: requests.models.Response HTTP response which can be used to stream
+            the payload of the response
+        :raises: HTTPError
+        """
+        if not id_:
+            raise ValueError('id_ must be provided')
```
Comment (amikofalvy) [inline src/rev_ai/apiclient.py]: do we need to note this `:raises: ValueError`? What is the convention?

File: src/rev_ai/apiclient.py [hunk 3]
```diff
@@ -195,19 +193,30 @@ def get_transcript_text(self, id_, filename=None, filepath=None):
...
+
+        url = urljoin(self.base_url, 'jobs/{}/transcript'.format(id_))
```
Comment (amikofalvy) [inline src/rev_ai/apiclient.py]:
```suggestion
        url = urljoin(self.base_url, 'jobs/{id}/transcript'.format(id_))
```

Are we supposed to put a key in the format bracket?

File: src/rev_ai/apiclient.py [hunk 4]
```diff
@@ -260,13 +280,26 @@ def get_captions(self, id_, filename=None, filepath=None):
...
+        :param id_: id of job to be requested
+        :returns: requests.models.Response HTTP response which can be used to stream
+            the payload of the response
+        :raises: HTTPError
+        """
+        if not id_:
+            raise ValueError('id_ must be provided')
+
+        url = urljoin(self.base_url, 'jobs/{}/captions'.format(id_))
+        response = self.session.get(
```
Comment (amikofalvy) [inline src/rev_ai/apiclient.py]: Should we do like we have done in the node sdk and separate out an api helper?

Comment (amikofalvy) [reply src/rev_ai/apiclient.py]: It seems like our python sdk (especially the tests suite) would benefit from the same design pattern.

File: src/rev_ai/apiclient.py [hunk 5]
```diff
@@ -260,13 +280,26 @@ def get_captions(self, id_, filename=None, filepath=None):
...
 
-        if filename:
-            path = os.path.join(filepath or '', filename+'.txt')
-            with open(path, 'w+') as f:
-                f.write(response.text)
-
         return response.text
 
+    def get_captions_as_stream(self, id_):
+        """Get the captions output of a specific job and return it as a plain text stream
```
Comment (amikofalvy) [inline src/rev_ai/apiclient.py]: Do we have an issue to add vtt output format to python SDK?

File: src/rev_ai/apiclient.py (hunk 3 above)
Comment (k-weng) [reply src/rev_ai/apiclient.py]: I believe it is not required by Python.

File: src/rev_ai/apiclient.py (hunk 2 above)
Comment (kbridbur) [reply src/rev_ai/apiclient.py]: It's not strictly necessary. My opinion is that it's a good convention to have just to make it clear what arguments are needed and have better error handling if a customer messes up and passes us something undefined

File: src/rev_ai/apiclient.py (hunk 3 above)
Comment (kbridbur) [reply src/rev_ai/apiclient.py]: I don't think so. All the other .formats work like this

File: src/rev_ai/apiclient.py (hunk 5 above)
Comment (kbridbur) [reply src/rev_ai/apiclient.py]: I will file one. It will likely be done along with AISDK-52 which will abstract away the requests package

**Pull request merged**